lib/topics_filter.rb

```diff
@@ -368,6 +370,19 @@ def tag_ids_from_tag_names(tag_names)
     all_tag_ids
   end
 
+  def tag_groups_to_tags(values)
+    values.map do |key_prefix, value|
+      tag_groups_names =
+        if value.include?(",")
+          value.split(",")
+        else
+          [value]
+        end
+      tags = TagGroup.where(name: tag_groups_names).map(&:tags).flatten.map(&:name).join(",")
```

Comment:
Instead of going to the database and fetching all the tags for the given tag groups, I think we should just join against the `tag_groups` and `tags` table instead we will be safer and more efficient when the given tag_groups have alot of tags.

Reply:
Like this? `TagGroup.where(name: tag_groups_names).joins(:tags).pluck(:name).join(",")` 

or with `<<~SQL`(heredoc)

Reply:
Sorry what I mean here is `tag_groups` can contain many tags and this makes the current SQL query risky cause we could potentially be fetching 10,000 tag names into memory if the tag group happens to contain 10,000 tags. 

As a result, I don't think we should rely on `filter_tags` here which requires all the tag names to be passed in and instead implement a `filter_tag_groups` method which doesn't have to fetch all the tag names into memory. 

One thing to take note of in this current PR is that you did not consider permissions of `tag_groups`. We need to  ensure that the user is allowed to view the specified tag groups.

Reply:
Another note here is that we have unit tests for `TopicsFilter` in 

https://github.com/discourse/discourse/blob/main/spec/lib/topics_filter_spec.rb

Whatever changes we add here should be accompanied by unit tests in the file above.